### DISCUSSION TITLE
ESP32 SD card support

### DISCUSSION THREAD

**Discussion by rubbaband1** [forkineye/ESPixelStick#409] [General]
Is this working with ESP32? I am testing with an mh-et live esp32 board (and another one made by someone with same exact pinout) which is a d1 mini style board with a second row of pinouts on each side. Using a D1 mini sd card breakout I have tested working on an esp8266 wemos d1 mini. When connected to the mh-et I flash the generic esp32 4mb profile. Shows no sd card installed after flashing. Enabling advanced and looking at the assigned pins it is showing chip select on IO4 but looking at the sd card breakout it should be going to IO5, so i switched that and saved, clk/miso/mosi were correct. Even after saving changes and rebooting still no SD card installed. Wondering if I am missing something somewhere?

**Comment by forkineye** [comment]
There are many people using the ESP32 Lolin D32 Pro and ESP32 CAM builds for SD support (I've tested both as well).  I haven't tested SD usage against the generic ESP32 build with the Wemos style units though.

**Comment by garymueller** [comment]
running 4.0-beta4: I was having the same issue using a ESP32 Wemos mini style with a sd card shield. 

![D1 Mini NodeMCU ESP32 ESP-WROOM-32](https://user-images.githubusercontent.com/35151679/152696923-606caeae-2558-4ca8-8e5d-66deb69ca1bf.jpg)

It appears that these ESP32's are expecting GPIO5 for the pin to be compatible with the shields.  Luckily I stumbled across this post and discovered advanced mode which allowed me to set the chip select pin to 5 and it worked for me.

Should the chip select pin be defaulted to 5 to be compatible with these boards and shields for people coming from the wemos mini's?
